src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -793,7 +793,7 @@ public void run() {
                                     result = oauthRequest.getToken(endingUrl);
                                     Logger.v(TAG, "OnActivityResult processed the result. "
                                             + authenticationRequest.getLogInfo());
-                                } catch (Exception exc) {
+                                } catch (IOException exc) {
```

Comment:
We should only be throwing IllegalArgumentException when doing parameter validation in public methods.

Catching AuthenticationException makes sense if it needs to be transferred to another thread.